src/scripts/records/VideoRecords.js

```diff
@@ -1,27 +1,36 @@
 /* @flow */
 
 import { Record as ImmutableRecord } from 'immutable'
+import AsyncTaskStatusRecord from './AsyncTaskStatusRecord'
 
 class Video extends ImmutableRecord({
+  blockchainStatus: new AsyncTaskStatusRecord(),
   description: '',
+  filename: null,
   id: '',
   ipfsData: '',
   ipfsHash: '',
   owner: '',
   price: '',
   thumbnailUrl: '',
   title: '',
+  transcodingStatus: new AsyncTaskStatusRecord(),
+  uploadStatus: new AsyncTaskStatusRecord(),
   url: ''
 }) {
-  description: string;
-  id: string;
-  ipfsData: string;
-  ipfsHash: string;
-  owner: string;
-  price: string;
-  thumbnailUrl: string;
-  title: string;
-  url: string;
+  blockchainStatus: AsyncTaskStatusRecord
+  description: string
+  filename: string
```

Comment:
One thing to look at is will these properties always be set? Some of them may need to be optionals if `paratii-lib` will not always include these fields (e.g. `id: string` => `id?: string` or `id: ?string`). The former is if the key is sometimes left out, and the latter is if the key is always there but the value might be `undefined` or `null`. You could even do `id?: ?string` if both are possible.